src/main/java/net/minecraftforge/event/entity/EntityFinalizeSpawnEvent.java

```diff
@@ -0,0 +1,55 @@
+package net.minecraftforge.event.entity;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.DifficultyInstance;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.MobSpawnType;
+import net.minecraft.world.entity.SpawnGroupData;
+
+/**
+ * This event is called when an entity finalizes its spawn, usually to give it something (eg, skeletons with bows)<br>
+ * This event should be used if a modder wants to add something to an entity the first time it's spawned.<br>
+ * If a modder wants to add something everytime an entity is spawned they should use {@link EntityJoinWorldEvent}
+ */
+public class EntityFinalizeSpawnEvent extends EntityEvent
+{
+    private final DifficultyInstance difficultyInstance;
+    private final MobSpawnType mobSpawnType;
+    @Nullable
+    private final SpawnGroupData groupData;
+    @Nullable
+    private final CompoundTag compoundTag;
+
+    public EntityFinalizeSpawnEvent(Mob entity, DifficultyInstance difficultyInstance, MobSpawnType mobSpawnType, @Nullable SpawnGroupData data, @Nullable CompoundTag compoundTag)
```

Comment:
In the interests of convenience of avoiding a cast/pattern-matching `instanceof`, please override the `getEntity` getter and change the return type to `Mob`.[^1]

Also, is this event intended for _any_ `Entity` while finalizing spawning, or only for `Mob`s? If only for `Mob`s, it may be beneficial to rename this event to `MobFinalizeSpawnEvent`, to clarify that this only fires for mobs. If not, then the type of the first parameter should be expanded to `Entity` to signal that it may not necessarily be a `Mob` stored in this event.

[^1]: This is possible because [return types are _covariant_](https://www.baeldung.com/java-covariant-return-type) in Java 5+ -- that is, an override can narrow down its return type to be a subclass of the overriden method's return type.